sdk/core/azure-core/src/datetime.cpp

```diff
@@ -610,13 +614,13 @@ DateTime DateTime::FromString(std::string const& dateString, DateFormat format)
     // parse month
     if (!IsDigit<1>(str[0]) || !IsDigit(str[1]))
     {
-      return result;
+      throw DateTimeException("Error parsing DateTime.");
```

Comment:
Consider adding contextual detail to all these error messages since we know where the parsing error occurred.